client/src/components/Workflow/Editor/Index.vue

```diff
@@ -354,7 +354,7 @@ export default {
             }
         }
 
-        const tags = ref([]);
+        const tags = ref(props.workflowTags);
```

Comment:
I think you'll be mutating the prop in https://github.com/galaxyproject/galaxy/blob/55439e455e81b7e900b21d4926495ac4abb58ac9/client/src/components/Workflow/Editor/Index.vue#L360.

```suggestion
        const tags = ref([...props.workflowTags]);
```
might work maybe ? I didn't test that but I think that shouldn't mutate the prop ?

Reply:
Both work but I can't see any difference here. BTW, I made the changes to prevent future issues.

Reply:
The linked line does not mutate the prop, since it overwrites the entire array, not its contents.

While initialisation of a ref via a prop does work, it breaks reactivity for the prop. This is currently a problem for multiple parts of the workflow editor, but it would be nice if we could get rid of them bit by bit. In order to initialise a ref with a prop, set it via a watcher set to immediate. That way when the prop changes from a parent component, the ref is also updated.